kotlin/display-feature-layers/src/main/java/com/esri/arcgisruntime/sample/displayfeaturelayers/MainActivity.kt

```diff
@@ -0,0 +1,263 @@
+        mapView.map = map
+        setUpBottomUI()
+    }
+
+    private fun setUpBottomUI() {
+        // create an adapter with the types of feature layer
+        // sources to be displayed in menu
+        val adapter = ArrayAdapter(
+            this,
+            android.R.layout.simple_list_item_1,
+            resources.getStringArray(R.array.feature_layer_sources)
+        )
+        bottomListItems.apply {
+            // populate the bottom list with the feature layer sources
+            setAdapter(adapter)
+            // click listener when feature layer source is selected
+            setOnItemClickListener { _, _, i, _ ->
+                // get the selected feature layer source
+                val selectedSource = FeatureLayerSource.values().find { it.menuPosition == i }
+                // check if the same feature is selected
+                if (previousSource != null && (previousSource == selectedSource)) {
+                    // same feature layer selected, return
+                    return@setOnItemClickListener
+                }
+                // set the feature layer source using the selected source
+                when (selectedSource) {
+                    FeatureLayerSource.SERVICE_FEATURE_TABLE -> loadFeatureServiceURL()
+                    FeatureLayerSource.PORTAL_ITEM -> loadPortalItem()
+                    FeatureLayerSource.GEODATABASE -> loadGeodatabase()
+                    FeatureLayerSource.GEOPACKAGE -> loadGeopackage()
+                    FeatureLayerSource.SHAPEFILE -> loadShapefile()
+                }
+                // update the previous feature layer source
+                previousSource = selectedSource
+            }
+        }
+    }
+
+    /**
+     * Load a feature layer with a URL
+     */
+    private fun loadFeatureServiceURL() {
+        // initialize the service feature table using a URL
+        val serviceFeatureTable =
+            ServiceFeatureTable(resources.getString(R.string.sample_service_url)).apply {
+                // set user credentials to authenticate with the service
+                credential = UserCredential("viewer01", "I68VGU^nMurF")
+                // NOTE: Never hardcode login information in a production application
+                // This is done solely for the sake of the sample
+            }
+        // create a feature layer with the feature table
+        val featureLayer = FeatureLayer(serviceFeatureTable)
+        val viewpoint = Viewpoint(41.773519, -88.143104, 4000.0)
+        // set the feature layer on the map
+        setFeatureLayer(featureLayer, viewpoint)
+    }
+
+    /**
+     * Load a feature layer with a portal item
+     */
+    private fun loadPortalItem() {
+        // set the portal
+        val portal = Portal("https://www.arcgis.com", false)
+        // create the portal item with the item ID for the Portland tree service data
+        val portalItem = PortalItem(portal, "1759fd3e8a324358a0c58d9a687a8578")
```

Comment:
could listen for error here too since you're doing it with the other feature layer source creations